From 6d7ce4d6e5ab8d4222bb41b4e09e3437d534481a Mon Sep 17 00:00:00 2001
From: Stephanos Ioannidis <root@stephanos.io>
Date: Wed, 26 Feb 2020 12:30:55 +0900
Subject: [PATCH 2/2] Enable WFI CPU halting in icount mode

This commit enables CPU halting via the WFI instruction in the
icount mode.

The PetaLinux patch initially disabled WFI CPU halting, possibly for
performance reasons.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>
---
 target/arm/op_helper.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/target/arm/op_helper.c b/target/arm/op_helper.c
index 6ab3a75a76..1199288fdd 100644
--- a/target/arm/op_helper.c
+++ b/target/arm/op_helper.c
@@ -304,12 +304,9 @@ void HELPER(wfi)(CPUARMState *env, uint32_t insn_len)
     }
 
     qemu_mutex_lock_iothread();
-    if (use_icount) {
-        cs->exception_index = EXCP_YIELD;
-    } else {
-        cs->halted = 1;
-        cs->exception_index = EXCP_HLT;
-    }
+
+    cs->halted = 1;
+    cs->exception_index = EXCP_HLT;
 
     /* Drive STANDBYWFI only if cpu reset-pin is inactive */
     if (cs->reset_pin == false) {
-- 
2.17.1

